activesupport/lib/active_support/error_reporter.rb

```diff
@@ -71,12 +71,33 @@ def initialize(*subscribers, logger: nil)
     # * +:source+ - This value is passed along to subscribers to indicate the
     #   source of the error. Subscribers can use this value to ignore certain
     #   errors. Defaults to <tt>"application"</tt>.
-    def handle(*error_classes, severity: :warning, context: {}, fallback: nil, source: DEFAULT_SOURCE)
-      error_classes = [StandardError] if error_classes.blank?
-      yield
-    rescue *error_classes => error
-      report(error, handled: true, severity: severity, context: context, source: source)
-      fallback.call if fallback
+    #
+    # * +:if+ - A boolean that determines whether +handle+ should be executed.
+    #   The block will be executed either way, but errors will only be handled
+    #   and reported if this value is +true+. For example:
+    #
+    #     Rails.error.handle(if: Rails.env.production?) do
+    #       raise "This error will only be raised and reported in production"
+    #     end
+    #
+    # * +:unless+ - A boolean that determines whether +handle+ should be
+    #   executed. The block will be executed either way, but errors will only be
+    #   handled and reported if this value is +false+. For example:
+    #
+    #     Rails.error.handle(unless: Rails.env.test?) do
+    #       raise "This error will only be raised during tests"
+    #     end
+    #
+    #   +:if+ and +:unless+ cannot be specified together.
+    def handle(*error_classes, severity: :warning, context: {}, fallback: nil, source: DEFAULT_SOURCE, **options)
+      yield unless continue?(options.slice(:if, :unless).compact)
```

Comment:
Should this be

```suggestion
      return yield unless continue?(options.slice(:if, :unless).compact)
```

I also think the method name isn't the clearest, what do you think about

```suggestion
      return yield unless use_error_handler?(options.slice(:if, :unless).compact)
```

Reply:
Thank you so much for this feedback. I'm going to take care of this and add a test case for not returning with the yield 🤦🏻